includes/purge-set-object.batch.inc

```diff
@@ -0,0 +1,130 @@
+  module_load_include('inc', 'islandora_batch', 'includes/db');
+  $batch = array(
+    'title' => t('Purge objects from set'),
+    'finished' => 'islandora_batch_batch_purge_object_in_set_finished',
+    'init_message' => t('Initializing...'),
+    'operations' => array(),
+    'progress_message' => t('Time elapsed: @elapsed.'),
+    'file' => drupal_get_path('module', 'islandora_batch') . '/includes/purge-set-object.batch.inc',
+  );
+
+  $batch['operations'][] = array('islandora_batch_batch_purge_object_in_set_operation', array('set_id' => $set_id));
+  batch_set($batch);
+}
+
+/**
+ * Batch operation to purge object ingested via batch.
+ *
+ * @param int $set_id
+ *   Batch Set ID.
+ *
+ * @param array $context
+ *   Batch context.
+ */
+function islandora_batch_batch_purge_object_in_set_operation($set_id, &$context) {
+  module_load_include('inc', 'islandora_batch', 'includes/ingest.batch');
+  module_load_include('inc', 'islandora_batch', 'includes/db');
+  $sandbox =& $context['sandbox'];
+  if (!islandora_batch_get_lock($set_id)) {
+    // Failed to acquire the lock... Try again in another iteration...
+    $context['finished'] = 0;
+    return;
+  }
+
+  if (!isset($sandbox['total'])) {
+    $sandbox['items'] = islandora_batch_get_queued_items_for_set($set_id);
+    dd($sandbox['items'], 'all items');
+    $sandbox['total'] = count($sandbox['items']);
+    $sandbox['progress'] = 0;
+    if ($sandbox['total'] === 0) {
+      $context['message'] = t('Nothing to process...');
+      return;
+    }
+  }
+  // Might be increased to avoid bootstrapping overhead.
+  $slice_size = 30;
+  $slice_stop = $sandbox['progress'] + $slice_size;
+
+  for (; $sandbox['progress'] < $slice_stop && $sandbox['progress'] < $sandbox['total']; $sandbox['progress']++) {
+    $purge = FALSE;
+    $item_id = $sandbox['items'][$sandbox['progress']];
+    $object = islandora_object_load($item_id);
+    if ($object) {
+      if ($object->repository->purgeObject($object->id)) {
+        $purge = TRUE;
+      }
+    }
+    if ($purge) {
+      $context['results']['purged'][] = $item_id;
+    }
+    else {
+      $context['results']['purged_fail'][] = $item_id;
+    }
+  }
+  $context['finished'] = $sandbox['progress'] / $sandbox['total'];
+}
+
+/**
+ * Finished function for deleting ingested objects.
+ *
+ * @param array $success
+ *   An array of success passed from the batch.
+ * @param array $results
+ *   An array of results passed from the batch.
+ * @param array $operations
+ *   An array of operations passed from the batch.
+ */
+function islandora_batch_batch_purge_object_in_set_finished($success, $results, $operations) {
+  if (isset($results['purged'])) {
+    $purged_count = count($results['purged']);
+    drupal_set_message(t('Purged @count @label.', array(
+      '@count' => $purged_count,
+      '@label' => format_plural($purged_count, 'object', 'objects'),
+    )), 'warning');
+  }
+  else {
+    drupal_set_message('No objects were purged', 'warning');
```

Comment:
Not 100% sure, but thinking coder would complain about this, also.  (`drupal_set_message()` with message not wrapped in `t()`/`filter_xss()`).